engine/app/lib/citizens_advice_components/elements/collections/base.rb

```diff
@@ -37,13 +37,23 @@ def hint_html
         def error_message_html
           return unless error?
 
-          tag.p(class: "cads-form-field__error-message") { error_message }
+          tag.p(class: "cads-form-field__error-message", id: error_id) { error_message }
         end
 
         def optional_html
           return unless optional
 
-          tag.span(class: "cads-form-field__optional") { "(optional)" }
+          tag.span(class: "cads-form-field__optional") do
+            "(#{I18n.t('citizens_advice_components.input.optional')})"
+          end
```

Comment:
Might be worth me pulling the actual fixes out into separate PRs so this change is just test coverage. Wanted to give an indication of the kinds of issues the tests spotted.